ultralytics/nn/modules/head.py

```diff
@@ -80,6 +79,101 @@ def bias_init(self):
             b[-1].bias.data[:m.nc] = math.log(5 / m.nc / (640 / s) ** 2)  # cls (.01 objects, 80 classes, 640 img)
 
 
+class DetectContrastive(nn.Module):
+    """YOLOv8 Detect head for detection models using Contrastive Learning."""
+    dynamic = False  # force grid reconstruction
+    export = False  # export mode
+    shape = None
+    anchors = torch.empty(0)  # init
+    strides = torch.empty(0)  # init
+
+    def __init__(self, nc=80, ch=()):  # ch is a list of number of channels in input layer (e.g., [ch_small, ch_medium, ch_large])
+        """Initializes the YOLOv8 detection layer with specified number of classes and channels.
+        This implementation has an additional contrastive learning task for embeddings of predicted bboxes.
+        This expands (doubles) the out_channels of the last convolution layer before classification layer 
+        to split embedding dimension for classification and contrastive learning tasks.
+        """
+        super().__init__()
+        self.nc = nc  # number of classes
+        self.nl = len(ch)  # number of detection layers (e.g., small, medium, large for 3-scale detections)
+        self.reg_max = 16  # DFL channels (ch[0] // 16 to scale 4/8/12/16/20 for n/s/m/l/x)
+        c2, c3 = max((16, ch[0] // 4, self.reg_max * 4)), max(ch[0], min(self.nc, 100))  # channels
```

Comment:
Extraneous parentheses.

Reply:
Removed the extra parentheses.
These extra parentheses came from the original Ultralytics code, so I removed these from `Detect()` class as well.
`c2, c3 = max((16, ch[0] // 4, self.reg_max * 4)), max(ch[0], min(self.nc, 100) --> c2, c3 = max(16, ch[0] // 4, self.reg_max * 4), max(ch[0], min(self.nc, 100)`